test/validator.test.ts

```diff
@@ -294,6 +321,17 @@ describe('Validator', () => {
     expect(result).toEqual([]);
   });
 
+  it('should validate constructor if configured', () => {
```

Comment:
I would split this test into 2 and move each of them to a separate `describe` block that redefines the config in a before hook.
```ts
describe('with enforced constructor natspec', () => {
  beforeAll(async () => {
    validator = new Validator(mockConfig({ constructorNatspec: true }));
  });

  it('should reveal missing constructor natspec', () => {
    // ...
  });
});

describe('with disabled constructor natspec', () => {
  beforeAll(async () => {
    validator = new Validator(mockConfig({ constructorNatspec: false }));
  });

  it('should ignore missing constructor natspec', () => {
    // ...
  });
});
```
This will be helpful if we plan to add more tests, for instance making sure that in case of `constructorNatspec: false` adding constructor natspec does not result in any warnings.